x-pack/test/functional/services/ml/expanded_job_details.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { MlADJobTable } from './job_table';
+
+export function MachineLearningExpandedJobDetailsProvider(
+  { getService }: FtrProviderContext,
+  jobTable: MlADJobTable
+) {
+  const testSubjects = getService('testSubjects');
+  const retry = getService('retry');
+  const find = getService('find');
+
+  return {
+    async assertCalendarPresent(jobId: string, expectedCalendar: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async function verifyCalendar() {
+        const calendarSelector = `${jobId}-${expectedCalendar}`;
+        await testSubjects.existOrFail(calendarSelector, {
+          timeout: 3_000,
+        });
+        const calendarVisibleText = await testSubjects.getVisibleText(calendarSelector);
+        expect(calendarVisibleText).to.be(expectedCalendar);
+      });
+    },
+
+    async assertForecastElements(jobId: string): Promise<void> {
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await this.openForecastTab(jobId);
+        await testSubjects.existOrFail('mlJobListForecastTabOpenSingleMetricViewButton', {
+          timeout: 3_000,
+        });
+      });
+    },
+
+    async openForecastTab(jobId: string) {
+      await testSubjects.click(jobTable.detailsSelector(jobId, 'mlJobListTab-forecasts'), 3_000);
+    },
+
+    async assertAnnotationsEdit(jobId: string): Promise<void> {
+      const newAnnotation = 'appex qa rox';
+      await jobTable.withDetailsOpen(jobId, async () => {
+        await jobTable.openAnnotationsTab(jobId);
```

Comment:
We should probably move the `openAnnotationsTab` method to the expanded job details service so it's consistent with the the other actions in an expanded job row. The same holds true for a number of other methods in the `job_table` service. However, we could do this cleanup in a follow-up PR.

Reply:
Will be done in a follow up pr